spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/integration/inbound/PubSubInboundChannelAdapter.java

```diff
@@ -80,11 +81,11 @@ private void receiveMessage(PubsubMessage pubsubMessage, AckReplyConsumer consum
 					.copyHeaders(messageHeaders)
 					.build());
 		}
-		catch (Exception re) {
+		catch (RuntimeException re) {
 			if (this.ackMode == AckMode.AUTO) {
 				consumer.nack();
 			}
-			throw new RuntimeException(re);
+			throw new PubSubException("Spring Message construction from Pub/Sub message failed.", re);
```

Comment:
Why "construction"?
I've explained in the previous comment that there can be anything downstream in the `sendMessage()`...

How transaction timeout is related to the "construction"? 🤷‍♂️